src/funnels/Depositor.sol

```diff
@@ -0,0 +1,293 @@
+// SPDX-FileCopyrightText: © 2020 Lev Livnev <lev@liv.nev.org.uk>
+// SPDX-FileCopyrightText: © 2021 Dai Foundation <www.daifoundation.org>
+// SPDX-License-Identifier: AGPL-3.0-or-later
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.8.16;
+
+interface RolesLike {
+    function canCall(bytes32, address, address, bytes4) external view returns (bool);
+}
+
+interface DepositedGemLike {
```

Comment:
Yeah this is because when all these various contracts define their own `GemLike`, contracts that import several of these contracts can't compile because of name conflicts. I thought at some point we might want to solve that by importing these interfaces from some `Interface.sol` or something but I wanted to avoid doing that just now while the design is still in flux. Though I guess it's a matter of taste, I also personally prefer to use slightly more awkward interface names than to import interfaces from an external file.

Reply:
mm I think you shouldn't have a naming a collision if you just import specficif contracts and interfaces and not full files.

Reply:
Oh yes that's right, changed.